tests/pipelines/stable_diffusion/test_stable_diffusion_controlnet.py

```diff
@@ -0,0 +1,574 @@
+
+    def test_attention_slicing_forward_pass(self):
+        return self._test_attention_slicing_forward_pass(expected_max_diff=2e-3)
+
+    @unittest.skipIf(
+        torch_device != "cuda" or not is_xformers_available(),
+        reason="XFormers attention is only available with CUDA and `xformers` installed",
+    )
+    def test_xformers_attention_forwardGenerator_pass(self):
+        self._test_xformers_attention_forwardGenerator_pass(expected_max_diff=2e-3)
+
+    def test_inference_batch_single_identical(self):
+        self._test_inference_batch_single_identical(expected_max_diff=2e-3)
+
+    def test_stable_diffusion_controlnet_ddim(self):
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+
+        vae_scale_factor = 8
+        components = self.get_dummy_components_for_controlnet()
+        sd_pipe = StableDiffusionControlNetPipeline(**components)
+        sd_pipe = sd_pipe.to(torch_device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs_for_controlnet(device)
+        output = sd_pipe(**inputs)
+        image = output.images
+
+        image_slice = image[0, -3:, -3:, -1]
+        # print("image_slice", image_slice)
+
+        assert image.shape == (1, 32 * vae_scale_factor, 32 * vae_scale_factor, 3)
+        expected_slice = np.array(
+            [0.47653976, 0.4843403, 0.46522307, 0.39793792, 0.454136, 0.4749748, 0.37724984, 0.4025603, 0.47651842]
+        )
+
+        assert np.abs(image_slice.flatten() - expected_slice).max() < 1e-2
+
+    def test_stable_diffusion_controlnet_ddim_two_prompts(self):
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+
+        vae_scale_factor = 8
+        components = self.get_dummy_components_for_controlnet()
+        sd_pipe = StableDiffusionControlNetPipeline(**components)
+        sd_pipe = sd_pipe.to(torch_device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs_for_controlnet(device, num_of_prompts=2)
+        output = sd_pipe(**inputs)
+        image = output.images
+
+        image_slice0 = image[0, -3:, -3:, -1]
+        image_slice1 = image[1, -3:, -3:, -1]
+
+        # print("image_slice0", image_slice0)
+        # print("image_slice1", image_slice1)
+
+        assert image.shape == (2, 32 * vae_scale_factor, 32 * vae_scale_factor, 3)
+
+        expected_slice0 = np.array(
+            [0.4394728, 0.46073985, 0.49796283, 0.52271855, 0.51414967, 0.5314792, 0.47262335, 0.47206822, 0.48990324]
+        )
+        expected_slice1 = np.array(
+            [0.5315275, 0.4819456, 0.4750305, 0.4453807, 0.44164768, 0.47079763, 0.40049344, 0.39453578, 0.47368276]
+        )
+
+        assert np.abs(image_slice0.flatten() - expected_slice0).max() < 1e-2
+        assert np.abs(image_slice1.flatten() - expected_slice1).max() < 1e-2
+
+    def test_stable_diffusion_controlnet_ddim_two_images_per_prompt(self):
```

Comment:
Good point! This is a test case we don't actually cover in the pipeline tester mixin, I'll include in a separate PR to cover all pipelines

Reply:
will add to mixin here: https://github.com/huggingface/diffusers/pull/2488